core/templates/components/entity-creation-services/collection-creation.service.ts

```diff
@@ -16,55 +16,67 @@
  * @fileoverview Modal and functionality for the create collection button.
  */
 
-// eslint-disable-next-line max-len
-require('components/entity-creation-services/collection-creation-backend-api.service.ts');
-require('services/alerts.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require('services/site-analytics.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Location } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-// TODO(bhenning): Refactor this to match the frontend design spec and reduce
-// duplicated code between CollectionCreationService and
-// ExplorationCreationService.
+import { AlertsService } from 'services/alerts.service';
+import { CollectionCreationBackendService, CollectionCreationResponse } from
+  'components/entity-creation-services/collection-creation-backend-api.service';
+import { LoaderService } from 'services/loader.service.ts';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
 
-angular.module('oppia').factory('CollectionCreationService', [
-  '$timeout', '$window', 'AlertsService',
-  'CollectionCreationBackendService', 'LoaderService', 'SiteAnalyticsService',
-  'UrlInterpolationService',
-  function(
-      $timeout, $window, AlertsService,
-      CollectionCreationBackendService, LoaderService, SiteAnalyticsService,
-      UrlInterpolationService) {
-    var CREATE_NEW_COLLECTION_URL_TEMPLATE = (
+@Injectable({
+  providedIn: 'root'
+})
+export class CollectionCreationService {
+  private collectionCreationInProgress: boolean = false;
```

Comment:
This should be static. Please see issue #9154 for an explanation.

Reply:
Fixed.